src/App/ElementNamingUtils.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <string>
+
+
+namespace Data
+{
+
+/// Special prefix to mark the beginning of a mapped sub-element name
+const std::string ELEMENT_MAP_PREFIX                = ";";
```

Comment:
Please see https://github.com/FreeCAD/FreeCAD/pull/8923#issuecomment-1476693963 and the link within it :D.

Reply:
Ah snap I've missed that. See also the discussion here https://github.com/FreeCAD/FreeCAD/commit/9af8e6a0556bd6def909a2787613ab0af698a06a#commitcomment-113554654.
Is there a better way to do this?

Reply:
Not sure of the variable naming convention. I would suggest staying with the rest of FreeCAD there.

Coming to the optimization, Pretty sure that would be asking too much. It could consider that a future file may import and want to use the string related functions, so it'll create the whole object rather than just the string. Perhaps that's why it's a strict no for libraries. This is all thought experiment though.